web/src/compose_recipient.ts

```diff
@@ -334,15 +372,21 @@ export function initialize() {
     });
 }
 
-export function update_placeholder_text() {
+export function update_placeholder_text(): void {
     // Change compose placeholder text only if compose box is open.
     if (!$("textarea#compose-textarea").is(":visible")) {
         return;
     }
+    const message_type = compose_state.get_message_type();
+    const stream_id = compose_state.stream_id();
+    assert(message_type !== undefined);
+    if (!stream_id) {
+        return;
+    }
```

Comment:
Why is it correct to return in this situation? Seems like you're introducing a bug when switching to DMs, but not sure.

Reply:
Using an assertion on `stream_id` causes an assertion failure when switching to DMs. I also don't think this is wrong because the `compute_placeholder_text` function in `compose_ui.ts` checks if the `message_type` is `stream` before trying to access the `stream_id` property. So this shouldn't introduce any bug in the code.

Reply:
I think this shouldn't be an assert, but it also shouldn't return -- this breaks the placeholder update for a DM, i.e. the `direct_message_user_ids` parameter, below.